backend/src/main/java/gov/cdc/usds/simplereport/service/DataHubUploaderService.java

```diff
@@ -4,70 +4,157 @@
+            JSONArray blocks = new JSONArray();
+            blocks.put(Map.of(
+                    "type", "header",
+                    "text", Map.of(
+                            "type", "plain_text",
+                            "text", titleMsg)
+            ));
+            markupMsgs.forEach(msg -> {
+                blocks.put(Map.of(
+                        "type", "section",
+                        "text", Map.of(
+                                "type", "mrkdwn",
+                                "text", msg)));
+                if (Boolean.TRUE.equals(separateMsgs)) {
+                    blocks.put(Map.of("type", "divider"));
+                }
+            });
+            String requestJson = new JSONObject(Map.of("blocks", blocks)).toString();
+
+            HttpEntity<String> entity = new HttpEntity<>(requestJson, headers);
+            restTemplate.put(_config.getSlackNotifyWebhookUrl(), entity);
+        } catch (RestClientException | JSONException err) {
+            LOG.error(err.toString());
+        }
+    }
+
+    // we put this in a function because the query can return null and it abstracts it out
+    private Date getLatestRecordedTimestamp() {
+        DataHubUpload lastUpload = _dataHubUploadRepo.findDistinctTopByJobStateOrderByLatestRecordedTimestampDesc(DataHubUpload.SUCCESS_JOB);
+        if (lastUpload != null) {
+            return lastUpload.getLatestRecordedTimestamp();
+        } else {
+            // This should only happen when database is empty
+            LOG.warn("Returning default timestamp - first run only");
+            return Date.from(Instant.parse("2020-01-01T00:00:00Z"));
+        }
+    }
+
+    // backwards compatible while refactoring
+    private void createTestEventCSV(String lastEndCreateOn) throws IOException, DateTimeParseException, NoResultException {
+        final Date DATE_1MIN_AGO = new Date(System.currentTimeMillis() - TimeUnit.MINUTES.toMillis(1));
         if (lastEndCreateOn.length() == 0) {
             // testing only, just query everything
             lastEndCreateOn = "2020-01-01T00:00:00.00Z";
             LOG.info("Empty startupdateby so using 2020-01-01");
         }
-        List<TestEvent> events = _repo.findAllByCreatedAtInstant(lastEndCreateOn);
+        createTestEventCSV(utcStringToDate(lastEndCreateOn), DATE_1MIN_AGO);
+    }
+
+    private void createTestEventCSV(Date earlistCreatedAt, Date latestCreateOn)
+            throws IOException, DateTimeParseException, NoResultException {
+        List<TestEvent> events = _testReportEventsRepo.findAllByCreatedAtBetweenOrderByCreatedAtDesc(earlistCreatedAt, latestCreateOn);
         if (events.size() == 0) {
             throw new NoResultException();
-        } else if (events.size() >= MAX_ROWS_ALLOWED_PER_CSV) {
+        } else if (events.size() >= _config.getMaxCsvRows()) {
             this._warnMessage += "More rows were found than can be uploaded in a single batch. Needs to be more than once.";
```

Comment:
Shouldn't we pass in a `LIMIT` instead? It looks right now like we're prepared to warn if it's too big, but then just try the upload anyway, which seems like it would go poorly for us.